demisto_sdk/commands/validate/validators/super_classes/BA106_is_from_version_sufficient.py

```diff
@@ -0,0 +1,132 @@
+from demisto_sdk.commands.content_graph.objects.incident_field import IncidentField
+from demisto_sdk.commands.content_graph.objects.incident_type import IncidentType
+from demisto_sdk.commands.content_graph.objects.indicator_field import IndicatorField
+from demisto_sdk.commands.content_graph.objects.job import Job
+from demisto_sdk.commands.content_graph.objects.layout import Layout
+from demisto_sdk.commands.content_graph.objects.layout_rule import LayoutRule
+from demisto_sdk.commands.content_graph.objects.list import List as LIST
+from demisto_sdk.commands.content_graph.objects.mapper import Mapper
+from demisto_sdk.commands.content_graph.objects.parsing_rule import ParsingRule
+from demisto_sdk.commands.content_graph.objects.playbook import Playbook
+from demisto_sdk.commands.content_graph.objects.pre_process_rule import PreProcessRule
+from demisto_sdk.commands.content_graph.objects.report import Report
+from demisto_sdk.commands.content_graph.objects.script import Script
+from demisto_sdk.commands.content_graph.objects.widget import Widget
+from demisto_sdk.commands.content_graph.objects.wizard import Wizard
+from demisto_sdk.commands.content_graph.objects.xsiam_dashboard import XSIAMDashboard
+from demisto_sdk.commands.content_graph.objects.xsiam_report import XSIAMReport
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    FixResult,
+    ValidationResult,
+)
+
+ContentTypes = Union[
+    GenericDefinition,
+    GenericField,
+    GenericModule,
+    GenericType,
+    LIST,
+    Mapper,
+    Classifier,
+    Widget,
+    Dashboard,
+    IncidentType,
+    Script,
+    Playbook,
+    Report,
+    Wizard,
+    Job,
+    Layout,
+    PreProcessRule,
+    CorrelationRule,
+    ParsingRule,
+    XSIAMDashboard,
+    XSIAMReport,
+    IncidentField,
+    IndicatorField,
+    AssetsModelingRule,
+    LayoutRule,
+]
+
+
+FROM_VERSION_DICT: Dict[ContentType, str] = {
+    ContentType.ASSETS_MODELING_RULE: "6.2.1",
+    ContentType.XSIAM_REPORT: "6.10.0",
+    ContentType.INCIDENT_FIELD: "5.0.0",
+    ContentType.INDICATOR_FIELD: "5.0.0",
+    ContentType.CORRELATION_RULE: "6.10.0",
+    ContentType.PARSING_RULE: "6.10.0",
+    ContentType.LAYOUT_RULE: "6.10.0",
+    ContentType.XSIAM_DASHBOARD: "6.10.0",
+    ContentType.SCRIPT: "5.0.0",
+    ContentType.PLAYBOOK: "5.0.0",
+    ContentType.REPORT: "5.0.0",
+    ContentType.WIZARD: "6.8.0",
+    ContentType.JOB: "6.8.0",
+    ContentType.LAYOUT: "6.0.0",
+    ContentType.PREPROCESS_RULE: "6.8.0",
+    ContentType.GENERIC_DEFINITION: "6.5.0",
+    ContentType.GENERIC_MODULE: "6.5.0",
+    ContentType.GENERIC_FIELD: "6.5.0",
+    ContentType.GENERIC_TYPE: "6.5.0",
+    ContentType.LIST: "6.5.0",
+    ContentType.MAPPER: "6.0.0",
+    ContentType.CLASSIFIER: "6.0.0",
+    ContentType.WIDGET: "5.0.0",
+    ContentType.DASHBOARD: "5.0.0",
+    ContentType.INCIDENT_TYPE: "5.0.0",
+}
+
+
+class IsFromVersionSufficientValidator(BaseValidator[ContentTypes]):
```

Comment:
lets make this 1 abstract - it only should have implementation details that are common to all children
Should be a standard that all classes not collected are abstract. Ideally this should be part of the infra